lib/utils/rruleUtil.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+let packageFound = false;
+try {
+  require('rrule');
+  packageFound = true;
+}
+catch (e) {
+  console.log('no package found');
+}
+
+let tryParseRRuleString = function(str) { return str;}
+let isRRuleObject = function(obj) {return false;}
+let nextInvocationDate = function(rrule, afterDate) {return null;}
+let doesRuleRecurr = function(rrule) {return false};
+
+
+if(packageFound) {
+  const {rrulestr, RRule, RRuleSet} = require('rrule');
+  /**
+   * Attempt to parse a string to a RRule object
+   * 
+   * @param {String} str the string to the parsed
+   * @returns A RRuleSet or RRule object if able to parse, otherwise will return the supplied string
+   */
+  tryParseRRuleString = function(str) {
+    try {
+      return rrulestr(str);
+    } catch (err) {
+      return str;
+    }
+  }
+
+  isRRuleObject = function(obj) {
+    return obj instanceof RRuleSet || obj instanceof RRule;
+  }
+
+  nextInvocationDate = function(rrule, afterDate) {
+    if(isRRuleObject(rrule)) {
```

Comment:
Do not swallow errors when we try to handle as RRule something that is not, throw a proper error

Reply:
Ok, Ill let it throw whatever error.
I was attempting to "protect" the user, but I suppose it could lead to confusing errors later down the line

Reply:
Yeah, it's better to catch userspace errors early by failing.